pkg/process_manager_test.go

```diff
@@ -107,3 +110,73 @@ func innerStartWaitStatus(t *testing.T, pm *ProcessManager) {
 	assert.Equal(EXITED, st.State)
 	assert.Equal(1, st.ExitCode)
 }
+
+func TestStream(t *testing.T) {
+	assert := assert.New(t)
+
+	pm := new(ProcessManager)
+	defer pm.StopAll()
+
+	// Ideally this test would use an IPC mechanism for synchronization, as time
+	// is not a synchronization primitive, and relying on timing for
+	// synchronization causes tests to become unnecessarily slow and unreliable.
```

Comment:
Ideally I'd also have a test for a reader joining partway through (in addition to from the beginning and after the end) but this is even more prone to being flaky without IPC-based synchronization so I'm going to skip on that I think.